app/admin/tenants/tenants.py

```diff
@@ -189,3 +198,25 @@ async def get_tenant(
     wallet = await aries_controller.multitenancy.get_wallet(wallet_id=tenant_id)
 
     return tenant_from_wallet_record(wallet)
+
+
+@router.get("/group/{group_id}", response_model=List[Tenant])
+async def get_tenants_by_group(
+    group_id: str, aries_controller: AcaPyClient = Depends(multitenant_admin)
+) -> List[Tenant]:
+    """Get tenants by group id."""
+    group_members = await actors_by_group_id(group_id=group_id)
+    wallets = await aries_controller.multitenancy.get_wallets()
+
+    if not wallets.results:
+        return []
+
+    group_members_ids = [member["id"] for member in group_members]
+    # Only return wallet with current group_id.
+    return [
+        tenant_from_wallet_record(
+            WalletRecordWithGroups(**wallet_record.dict(), group_id=group_id)
+        )
+        for wallet_record in wallets.results
+        if wallet_record.wallet_id in group_members_ids
+    ]
```

Comment:
There is a separate endpoint to query by the group id. The reason for adding the wallet groups is that it allows querying only the records in a group. This will still fetch all records and query locally, bypassing the performance gain